bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -385,8 +385,8 @@ public int process(boolean shouldForceWrite) throws IOException {
                     this.logFile.forceWrite(false);
                     journalStats.getJournalSyncStats()
                         .registerSuccessfulEvent(MathUtils.elapsedNanos(startTime), TimeUnit.NANOSECONDS);
+                    lastLogMark.setCurLogMark(this.logId, this.lastFlushedPosition);
                 }
-                lastLogMark.setCurLogMark(this.logId, this.lastFlushedPosition);
```

Comment:
I think that we can reach this point in some cases when there is no need to `force`

for instance here
https://github.com/apache/bookkeeper/blob/63867a99bac7df6e41b71e55af06ac514bd43113/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java#L524

Reply:
do you see any code path in which we arrive here and the journal has not been flushed and fsync'd ?